File: Algorithms/C/conversions/binary_to_decimal.c
#include <stdio.h>
#include <math.h>
#include <inttypes.h>

// uint64_t is an unsigned 64-bit integer

uint64_t binary_to_decimal(uint64_t n)
{
    uint64_t decimal = 0;
    uint64_t base = 1;
    uint64_t temp = n;
    while (temp > 0)
    {
        uint64_t last_digit = temp % 10;
        temp = temp / 10;
        decimal += last_digit * base;
        base = base * 2;
    }
    return decimal;
}

int main()
{
    uint64_t n = 1010;
    printf("Binary: %" PRIu64 "\n", n);
    printf("Decimal: %" PRIu64 "\n", binary_to_decimal(n));
    return 0;
}

File: Algorithms/Python/conversion/binary_to_decimal.py
from __future__ import annotations


def binary_to_decimal(binary: str) -> int:
    # return int(binary, 2)

    decimal = 0

    length = len(binary)
    for i in range(length):
        decimal += int(binary[i]) * 2 ** (length - i - 1)

    return decimal


if __name__ == "__main__":
    binary = "1010"
    print(binary_to_decimal(binary))  # 10
    binary = "1101"
    print(binary_to_decimal(binary))  # 13
    binary = "11111111"
    print(binary_to_decimal(binary))  # 255
    binary = "10000000000000000000000000000000"
    print(binary_to_decimal(binary))  # 2147483648
